### ISSUE TITLE
"webserver 12 /web-server/local-address-only" often fails in Rawhide

### ISSUE THREAD

**Issue by martinpitt** [cockpit-project/cockpit#11121]
Release 187 failed to build in rawhide [two](https://koji.fedoraproject.org/koji/taskinfo?taskID=32575591) [times](https://koji.fedoraproject.org/koji/taskinfo?taskID=32576309). 

In all cases, the failure looks like this:
```
cockpit-protocol:ERROR:src/common/test-webserver.c:353:perform_http_request: assertion failed (error == NULL): Could not connect to localhost: Connection refused (g-io-error-quark, 39)
FAIL: test-webserver 12 /web-server/local-address-only
```

**Comment by martinpitt** [comment]
I tried to reproduce this in a local rawhide mock:

    while ./test-webserver -p /web-server/local-address-only; do :; done

this works fine, but when running *all* tests, it's reproducible:
```
$ while ./test-webserver; do :; done
[...]
/web-server/local-address-only: **
cockpit-protocol:ERROR:src/common/test-webserver.c:353:perform_http_request: assertion failed (error == NULL): Could not connect to localhost: Connection refused (g-io-error-quark, 39)
Aborted (core dumped)
```

This does not happen in Fedora 29, the test is rock solid there. So something did change other than just bad timing.

**Comment by martinpitt** [comment]
It actually does reproduce with just the single test (`-p /web-server/local-address-only`), it just takes a little longer.

It is not a regression in glib-networking -- when running against 2.58.0 (as in Fedora 29), the bug still happens. But it does not happen any more when running against glib 2.58.0 (in an otherwise Rawhide environment):
```
fedora-29$ cp `rpm -ql glib2 | grep /usr/lib64/lib.* ` ~/tmp/
<mock-chroot> sh-4.4$ while LD_LIBRARY_PATH=/home/martin/tmp ./test-webserver; do :; done
```

**Comment by martinpitt** [comment]
I did a git bisect  with the run script below, for `git bisect start master 2.58.0`.

```sh
#!/bin/sh
set -eu

git clean -ffdx
meson b
ninja -C b

export LD_LIBRARY_PATH=b/glib:b/gobject:b/gthread:b/gmodule:b/gio
export GIO_MODULE_DIR=/usr/lib64/gio/modules/

for i in `seq 500`; do
    /home/martin/upstream/cockpit/test-webserver || exit 1
done
```

I filed https://gitlab.gnome.org/GNOME/glib/issues/1680 about the regression.

**Comment by martinpitt** [comment]
The fix is in glib 2.59.3 which just landed in Rawhide: https://koji.fedoraproject.org/koji/buildinfo?buildID=1211823
